src/query/storage/cache/utils.go

```diff
@@ -98,22 +98,57 @@ func samplesEqual(a, b []prompb.Sample, end int64, logger *zap.Logger) bool {
 	return true
 }
 
+// Create new result that doesn't include data past end
+func createEndFilteredTimeseries(timeseries []*prompb.TimeSeries, end int64) []*prompb.TimeSeries {
+	nonempty := make([]*prompb.TimeSeries, 0)
+	if timeseries != nil {
+		for _, ts := range timeseries {
+			// Check if all data is after time end
+			// If it is, then don't add it (i.e. drop timeseries as a whole)
+			if ts.Samples[0].Timestamp < end {
+				nonempty = append(nonempty, &prompb.TimeSeries{
+					Labels:  ts.Labels,
+					Samples: ts.Samples,
+				})
+			}
+		}
+	}
+	return nonempty
+}
+
+// Log the list of timeseries labels for a timeseries as list of strings
+func TimeseriesPrint(a, b []*prompb.TimeSeries, logger *zap.Logger) {
+	arr1 := make([]string, 0)
+	for i := range a {
+		arr1 = append(arr1, (&prompb.Labels{Labels: a[i].Labels}).String())
+	}
+	arr2 := make([]string, 0)
+	for i := range b {
+		arr2 = append(arr2, (&prompb.Labels{Labels: b[i].Labels}).String())
+	}
+	logger.Info("Timeseries difference", zap.Strings("actual labels", arr1), zap.Strings("expected labels", arr2))
+}
+
 // Check if lists of timeseries are equal (see samplesEqual for usage of {end} param)
 func TimeseriesEqual(a, b []*prompb.TimeSeries, end int64, logger *zap.Logger) bool {
-	if len(a) != len(b) {
-		logger.Info("Timeseries lengths unequal", zap.Int("actual length", len(a)), zap.Int("expected length", len(b)))
+	// Account for case where a new timeseries has just been recently added in the time difference
+	// In this case, filter out that data and recompare
+	a_filter := createEndFilteredTimeseries(a, end)
+	b_filter := createEndFilteredTimeseries(b, end)
```

Comment:
Actually when we skip the up to date minute to do the check, would be new metrics still concern?

Reply:
So the first check I make is that the number of metrics is the same so this is just a way to work around that check in case we add a new metric

Reply:
But we already skip the first minute right? If we don't check the first minute, will we have new metrics?

Reply:
So the part where I disregard the first minute comes later in that code. But before that I added something to check that the number of metrics are equal since this made comparison easier. The problem is that if a new metric only exists in that first minute for the newer request, it won't pass this metric number check and so it will fail before I do the check that skips over the first minute. This is just a way to make sure that this case makes it past that check for equal metric counts